help/develop/javascript-development/packages.md

```diff
@@ -0,0 +1,263 @@
+
+```shell
+npm install datagrok-tools -g
+```
+
+2. Configure your environment:
+
+```shell
+grok config
+```
+
+You will be prompted to enter the developer keys and set the default server. Your credentials will be stored locally
+in `config.yaml`. Check out the [Datagrok configuration section] for more details.
+
+3. Go to the folder where you want to create your package and run:
+
+```shell
+grok create <package-name>
+```
+
+> **Note**: If you want to create a TypeScript package, pass the `--ts` option to the command.
+
+After you complete the work on your Datagrok package, upload it to Datagrok by running the following command:
+
+```shell
+grok publish
+```
+
+Note that by running this command, only you will see the package in Datagrok. For more information about the available
+commands, refer to the [Grok CLI section].
+
+## Package structure
+
+| File                                    | Description                                               |
+|-----------------------------------------|---------------------------------------------------------- |
+| node_modules                            | Node.js modules                                           |
+| [src/package.js](#package.js)           | The Datagrok package's entry point                        |
+| .gitignore                              | Files and folders ignored by Git                          |
+| [detectors.js](#detectors)              | File with detectors                                       |
+| [package.json](#package.json)           | Package metadata with predefined dependencies and scripts |
+| package.png                             | Package icon                                              |
+| README.md                               | Summary of the package                                    |
+| [webpack.config.js](#webpack.config.js) | webpack configuration for development                     |
+
+We discuss the key files in detail in their own sections that follow.
+
+### <a href="#" id="package.json"></a>package.json
+
+The `package.json` file contains the project metadata.
+
+```json
+{
+  "name": "test-package",
+  "beta": true,
+  "friendlyName": "test-package",
+  "version": "0.0.1",
+  "description": "",
+  "dependencies": {
+    "datagrok-api": "latest"
+  },
+  "devDependencies": {
+    "webpack": "latest",
+    "webpack-cli": "latest"
+  },
+  "scripts": {
+    "debug-test-package": "grok publish --rebuild",
+    "release-test-package": "grok publish --rebuild --release",
+    "build-test-package": "webpack",
+    "build": "webpack",
+    "debug-test-package-dev": "grok publish dev --rebuild",
+    "release-test-package-dev": "grok publish dev --rebuild --release",
+    "debug-test-package-local": "grok publish local --rebuild",
+    "release-test-package-local": "grok publish local --rebuild --release"
+  }
+}
+```
+
+The package includes only the `datagrok-api` dependency &mdash; this is the JavaScript API you will actively use when
```

Comment:
This is no longer the case. Now the template has the following dependencies from the outset:
```json
"dependencies": {
    "datagrok-api": "latest",
    "cash-dom": "latest",
    "dayjs": "latest"
  },
  "devDependencies": {
    "webpack": "latest",
    "webpack-cli": "latest"
  },
```

Reply:
Let's assume our readers aren't necessarily familiar with the JS development ecosystem. Commands, such as `npm install` and `webpack`, should be explicitly mentioned in the documentation.